### PR TITLE
Fixed slider value indicator not disappearing after a bit on desktop platform when slider is clicked not dragged

### LINKED ISSUE #123313: Slider label never goes away when clicking slider to new value
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Click on the slider instead of dragging it
3. Note that slider label never goes away.  Tested on Windows 10 and Android, but I only see the issue on Windows.

**Expected results:** The label should go away after some duration, preferably a configurable one.

**Actual results:** The label never goes away.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const SliderApp());

class SliderApp extends StatelessWidget {
  const SliderApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: SliderExample(),
    );
  }
}

class SliderExample extends StatefulWidget {
  const SliderExample({super.key});

  @override
  State<SliderExample> createState() => _SliderExampleState();
}

class _SliderExampleState extends State<SliderExample> {
  double _currentSliderValue = 20;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Slider')),
      body: Slider(
        value: _currentSliderValue,
        max: 100,
        divisions: 5,
        label: _currentSliderValue.round().toString(),
        onChanged: (double value) {
          setState(() {
            _currentSliderValue = value;
          });
        },
      ),
    );
  }
}
```

</details>

### PR DESCRIPTION
**Pull request by cruiser-baxter**
In slider.dart within the _startInteraction method and within the below conditional.

"if (!_active && !hasFocus &&
_state.valueIndicatorController.status == AnimationStatus.completed)"

**Changed to:**

"f (!_active &&
_state.valueIndicatorController.status == AnimationStatus.completed)"
This allows the value indicator to disappear after a bit when clicked instead of dragged on Desktop platform. 

I also added a test in slider_test.dart to detect the bug if it ever returns.

Fixes https://github.com/flutter/flutter/issues/123313

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by TahaTesser — CHANGES_REQUESTED** [review]
Thanks for fixing this! 
A few changes are needed then it should be good to go.

**Review by TahaTesser — CHANGES_REQUESTED** [review]
Thanks for fixing this! 
A few changes are needed then it should be good to go.

**Comment by cruiser-baxter** [comment]
@TahaTesser, I sincerely appreciate your review and suggestions. After going over the diff for my own learning I committed the suggested changes. I learned a lot from the simplified test. Thank you.

**Review by TahaTesser — APPROVED** [review]
LGTM!

**Comment by TahaTesser** [comment]
This will need a second review from @Piinks to land.

**Review by Piinks — COMMENTED** [review]
Hey @cruiser-baxter, welcome! Thanks for contributing!
The description and test sound like this is specific to desktop, does this change alter the behavior on mobile? Will the label disappear now on mobile if there is focus? Might be good to add a test for this (if one does not already exist)
Looks like the `!hasFocus` was added in https://github.com/flutter/flutter/pull/113543 by @TahaTesser, which was also desktop specific.

**Comment by TahaTesser** [comment]
> Hey @cruiser-baxter, welcome! Thanks for contributing! The description and test sound like this is specific to desktop, does this change alter the behavior on mobile? 

This doesn't change the behavior on mobile. `_endInteraction` always dismisses the value indicator. 


> Looks like the !hasFocus was added in https://github.com/flutter/flutter/pull/113543 by @TahaTesser, which was also desktop specific.

Indeed I added it but based on the issue report, I found that interacting directly should dismiss the value indicator.  This issue fixes that interaction behavior.

While on the desktop the value indicator appears but doesn't dismiss when only using the keyboard. 
However current Slider implementation doesn't support invoking value indicators this way.  https://github.com/flutter/flutter/issues/113538. 

> Might be good to add a test for this (if one does not already exist)

Actually, we can remove the desktop target on this test. Since the interaction behavior is the same on desktop and mobile.

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -3507,6 +3507,54 @@ void main() {
...
+
+    // Wait for the value indicator to disappear.
+    await tester.pumpAndSettle(const Duration(seconds: 2));
+
+    // Value indicator is no longer visible.
+    expect(
+      valueIndicatorBox,
+      isNot(paints..path(color: const Color(0xff000000))..paragraph()),
+    );
+  }, variant: TargetPlatformVariant.desktop());
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3556]: We don't need this.

```suggestion
  });
```

File: packages/flutter/test/material/slider_test.dart [hunk 2]
```diff
@@ -3507,6 +3507,54 @@ void main() {
     );
   }, variant: TargetPlatformVariant.desktop());
 
+  testWidgets('Value indicator disappears after adjusting the slider', (WidgetTester tester) async {
+    // This is a regression test for https://github.com/flutter/flutter/issues/123313.
+    const double currentValue = 0.5;
+    await tester.pumpWidget(MaterialApp(
+      home: Material(
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3514]: To make this test future-proof. Let's Material 3 color to check value indicator visibility.
We won't need to update this later. 

```suggestion
    // This is a regression test for https://github.com/flutter/flutter/issues/123313.
    final ThemeData theme = ThemeData(useMaterial3: true);
    const double currentValue = 0.5;
    await tester.pumpWidget(MaterialApp(
      theme: theme,
      home: Material(
```

File: packages/flutter/test/material/slider_test.dart [hunk 3]
```diff
@@ -3507,6 +3507,54 @@ void main() {
...
+        ),
+      ),
+    ));
+
+    // Slider does not show value indicator initially.
+    await tester.pumpAndSettle();
+    RenderBox valueIndicatorBox = tester.renderObject(find.byType(Overlay));
+    expect(
+      valueIndicatorBox,
+      isNot(paints..path(color: const Color(0xff000000))..paragraph()),
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3531]:
```suggestion
      isNot(paints..path(color: theme.colorScheme.primary)..paragraph()),
```

File: packages/flutter/test/material/slider_test.dart [hunk 4]
```diff
@@ -3507,6 +3507,54 @@ void main() {
...
+
+    // Tap the slider to bring up the value indicator.
+    await tester.tapAt(tapLocation);
+    await tester.pumpAndSettle();
+
+    // Value indicator is visible.
+    valueIndicatorBox = tester.renderObject(find.byType(Overlay));
+    expect(
+      valueIndicatorBox,
+      paints..path(color: const Color(0xff000000))..paragraph(),
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3545]:
```suggestion
      paints..path(color: theme.colorScheme.primary)..paragraph(),
```

File: packages/flutter/test/material/slider_test.dart [hunk 5]
```diff
@@ -3507,6 +3507,54 @@ void main() {
...
+    );
+
+    // Wait for the value indicator to disappear.
+    await tester.pumpAndSettle(const Duration(seconds: 2));
+
+    // Value indicator is no longer visible.
+    expect(
+      valueIndicatorBox,
+      isNot(paints..path(color: const Color(0xff000000))..paragraph()),
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3554]:
```suggestion
      isNot(paints..path(color: theme.colorScheme.primary)..paragraph()),
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Review by cruiser-baxter — COMMENTED** [review]
I have reviewed the suggested changes and committed all suggestions. Thank you TahaTesser for all of your input, feedback, and code change suggestions.

**Review by Piinks — COMMENTED** [review]
Thank you for all the context @TahaTesser!

@cruiser-baxter it looks like there are some failing tests here now, can you take a look?

`/b/s/w/ir/x/w/flutter/packages/flutter/test/material/slider_test.dart: Value indicator disappears after adjusting the slider`

File: packages/flutter/test/material/slider_test.dart [hunk 6]
```diff
@@ -3507,6 +3507,56 @@ void main() {
...
+      paints..path(color: theme.colorScheme.primary)..paragraph(),
+    );
+
+    // Wait for the value indicator to disappear.
+    await tester.pumpAndSettle(const Duration(seconds: 2));
+
+    // Value indicator is no longer visible.
+    expect(
+      valueIndicatorBox,
+      isNot(paints..path(color: theme.colorScheme.primary)..paragraph()),
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3556]:
```suggestion
      isNot(paints..scale()..path(color: theme.colorScheme.primary)),
```

File: packages/flutter/test/material/slider_test.dart [hunk 7]
```diff
@@ -3507,6 +3507,56 @@ void main() {
...
+
+    // Tap the slider to bring up the value indicator.
+    await tester.tapAt(tapLocation);
+    await tester.pumpAndSettle();
+
+    // Value indicator is visible.
+    valueIndicatorBox = tester.renderObject(find.byType(Overlay));
+    expect(
+      valueIndicatorBox,
+      paints..path(color: theme.colorScheme.primary)..paragraph(),
```
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:3547]:
```suggestion
      paints..scale()..path(color: theme.colorScheme.primary),
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]
Just tried the test locally. After updating the test for M3 the expected order for `path` has changed.

**Review by TahaTesser — APPROVED** [review]
LGTM

**Comment by cruiser-baxter** [comment]
@TahaTesser Thank you for fixing that. I am in the middle of moving now and won't be able to look at this much for the next week.

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
Thank you!

**Pull request merged**